test/php/gutenberg-coding-standards/Gutenberg/Sniffs/Commenting/FunctionCommentSinceTagSniff.php

```diff
@@ -48,90 +91,444 @@ public function process( File $phpcsFile, $stackPtr ) {
 			return;
 		}
 
-		$tokens        = $phpcsFile->getTokens();
-		$function_name = $phpcsFile->getDeclarationName( $stackPtr );
+		$tokens = $phpcsFile->getTokens();
+		$token  = $tokens[ $stackPtr ];
+
+		if ( 'T_FUNCTION' === $token['type'] ) {
+			$this->process_function_token( $phpcsFile, $stackPtr );
+			return;
+		}
+
+		if ( isset( static::$oo_tokens[ $token['code'] ] ) ) {
+			$this->process_oo_token( $phpcsFile, $stackPtr );
+			return;
+		}
 
-		$wrapping_tokens_to_check = array(
-			T_CLASS,
-			T_INTERFACE,
-			T_TRAIT,
+		if ( 'T_STRING' === $token['type'] ) {
+			$this->process_hook( $phpcsFile, $stackPtr );
+			return;
+		}
+
+		if ( 'T_VARIABLE' === $token['type'] && Scopes::isOOProperty( $phpcsFile, $stackPtr ) ) {
+			$this->process_property_token( $phpcsFile, $stackPtr );
+		}
+	}
+
+	/**
+	 * Processes a token representing a function call that invokes a WordPress hook,
+	 * checking for a missing `@since` tag in its docblock.
+	 *
+	 * @param File $phpcs_file    The file being scanned.
+	 * @param int  $stack_pointer The position of the hook token in the stack.
+	 */
+	protected function process_hook( File $phpcs_file, $stack_pointer ) {
+		$tokens = $phpcs_file->getTokens();
+
+		// The content of the current token.
+		$hook_function = $tokens[ $stack_pointer ]['content'];
+
+		$hook_invocation_functions = array(
```

Comment:
Should `do_action_deprecated()` and `apply_filters_deprecated()` be supported as well? I'm not sure if the `@since` tag is required when calling those functions or not, but I see it is used in some calls that I randomly checked in the WP repository. I did not find any calls to those functions in the GB repository.

Reply:
I removed support for these deprecated functions at one point, but I agree, it's better to have these hooks covered. Let's add them back. Fixed in aed56d25f9890b7667c15241aee160009485e005

Reply:
I remember using them in the Gutenberg plugin at some point, so it's great idea to cover them 👍🏻

Reply:
Thanks for adding them back!